app/controllers/application_controller.rb

```diff
@@ -65,8 +65,10 @@ def render_404
   private
 
   def update_current_team
-    if current_user.current_team_id.blank? &&
-       current_user.teams.count > 0
+    current_team = Team.find_by_id(current_user.current_team_id)
+    if (current_team.nil? || !current_user.is_member_of_team?(current_team)) &&
+       current_user.teams.count.positive?
+
```

Comment:
Shouldn't this much logic be in a model/service object?

Reply:
While I agree, this should be done as a part of a refactoring task. Which we do have planned, in the future.